pkg/nodeconfig/nodeconfig.go

```diff
@@ -505,22 +509,26 @@ func (nc *nodeConfig) Deconfigure() error {
 		return errors.Wrapf(err, "unable to drain node %s", nc.node.GetName())
 	}
 
-	// Revert the changes we've made to the instance by removing services and deleting all installed files
-	if err := nc.Windows.Deconfigure(); err != nil {
-		return errors.Wrap(err, "error deconfiguring instance")
+	if err := nc.Windows.DeconfigureWICD(); err != nil {
+		return err
 	}
 
-	// Clear the version annotation from the node object to indicate the node is not configured
-	patchData, err := metadata.GenerateRemovePatch([]string{}, []string{metadata.VersionAnnotation})
-	if err != nil {
-		return errors.Wrapf(err, "error creating version annotation remove request")
+	// Revert the changes we've made to the instance by removing services and deleting all installed files
+	if err := nc.Windows.Deconfigure(preserveNode, nodeConfigCache.apiServerEndpoint); err != nil {
+		return errors.Wrap(err, "error deconfiguring instance")
 	}
-	_, err = nc.k8sclientset.CoreV1().Nodes().Patch(context.TODO(), nc.node.GetName(), kubeTypes.JSONPatchType,
-		patchData, meta.PatchOptions{})
-	if err != nil {
-		return errors.Wrapf(err, "error removing version annotation from node %s", nc.node.GetName())
+	if preserveNode {
+		// Clear the version annotation from the node object to indicate the node is not configured
+		patchData, err := metadata.GenerateRemovePatch([]string{}, []string{metadata.VersionAnnotation})
+		if err != nil {
+			return errors.Wrapf(err, "error creating version annotation remove request")
+		}
+		_, err = nc.k8sclientset.CoreV1().Nodes().Patch(context.TODO(), nc.node.GetName(), kubeTypes.JSONPatchType,
+			patchData, meta.PatchOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "error removing version annotation from node %s", nc.node.GetName())
+		}
```

Comment:
Shouldn't this also be done by WICD given it is the entity that decides whether to delete the node or node based on the `preservedNode` option?

Reply:
While this should be done, it might be out of scope of the PR. See [WINC-741](https://issues.redhat.com/browse/WINC-741)

Reply:
Ah yes, I keep forgetting about that story.